src/main/java/io/phdata/streamliner/schemadefiner/ConfigBuilder/DiffGenerator.java

```diff
@@ -0,0 +1,186 @@
+                                  .map(col -> col.getSourceName())
+                                  .collect(Collectors.toList());
+                          // current config all columns list
+                          List<String> currColList =
+                              currColDefList.stream()
+                                  .map(col -> col.getSourceName())
+                                  .collect(Collectors.toList());
+
+                          List<String> deletedColumns = getListDiff(prevColList, currColList);
+                          List<String> addedColumns = getListDiff(currColList, prevColList);
+                          List<String> commonColumns = getCommonCol(prevColList, currColList);
+                          List<String> updatedColumns = new ArrayList<>();
+
+                          deletedColumns.stream()
+                              .forEach(
+                                  deletedCol -> {
+                                    ColumnDefinition prevColDef =
+                                        prevColDefList.stream()
+                                            .filter(
+                                                colDef -> colDef.getSourceName().equals(deletedCol))
+                                            .collect(Collectors.toList())
+                                            .get(0);
+                                    ColumnDiff colDiff =
+                                        new ColumnDiff(prevColDef, null, true, false, false);
+                                    columnDiffs.add(colDiff);
+                                  });
+
+                          addedColumns.stream()
+                              .forEach(
+                                  addedCol -> {
+                                    ColumnDefinition currColDef =
+                                        currColDefList.stream()
+                                            .filter(
+                                                colDef -> colDef.getSourceName().equals(addedCol))
+                                            .collect(Collectors.toList())
+                                            .get(0);
+                                    ColumnDiff colDiff =
+                                        new ColumnDiff(null, currColDef, false, true, false);
+                                    columnDiffs.add(colDiff);
+                                  });
+
+                          commonColumns.stream()
+                              .forEach(
+                                  commonCol -> {
+                                    ColumnDefinition prevColDef =
+                                        prevColDefList.stream()
+                                            .filter(
+                                                colDef -> colDef.getSourceName().equals(commonCol))
+                                            .collect(Collectors.toList())
+                                            .get(0);
```

Comment:
This seems like it'd be much simpler in a map by sourceName? LinkedHashMap is a map which keeps insertion order iteration.

Reply:
yes it'd be much simpler. But since our model class is like this we have to play with list.

Reply:
Can't we create a LinkedHashMap and then store the source name as the key and the object as a value?

Reply:
yes we can do that. I will make the changes.